### ISSUE TITLE
Add a "validation only" mode

### ISSUE THREAD

**Issue by reinink** [brandonsavage/Upload#54]
This library is great, and the new [Flysystem integration](https://github.com/codeguy/Upload/pull/46) looks awesome, but I think it would be worthwhile to make it possible to use Upload without any file system interaction. Something like this:

``` php
// Note the lack of $storage parameter
$file = new \Upload\File('foo');

// Add validation rules
$file->addValidations(array(
    //...
));

// Validate the uploaded file
try {
    // Success!
    $file->validate();

    // Manually move the file (ie. S3)

} catch (\Exception $e) {
    // Fail!
    $errors = $file->getErrors();
}
```

I don't think this would take much to do, and is maybe an appropriate change after the Flysystem integration has been completed.

**Comment by MacFJA** [comment]
The [Flysystem](https://github.com/thephpleague/flysystem) library provide a `NullAdapter` which will do nothing in the Upload context.

You can define the `\Upload\File` with this adapter:
```php
use \League\Flysystem\Filesystem;
use \League\Flysystem\Adapter\NullAdapter;
$file = new \Upload\File('foo', new Filesystem(new NullAdapter));
//...
```
(I agree, it's a lot of writing for doing nothing)

Or the `NullAdapter` can be the default Flysystem Filesystem (IMHO is not a good idea, for me a default value must be the more generic one)

However, if you don't call the `upload` method, the file will not be moved.

P.S. The Flysystem library [have an adapter for S3](https://github.com/thephpleague/flysystem/blob/master/src/Adapter/AwsS3.php)

**Comment by reinink** [comment]
Yeah, not sure I want to use a Null adapter just to get around this, but thanks either way. :)

**Comment by brandonsavage** [comment]
What would be the use case here?

**Comment by reinink** [comment]
Hey Brandon, mostly just to allow using other file systems. The current implementation means saving the temporary file to another temporary location, just to move it to say S3, or otherwise. I posted this ticket specifically thinking about Flysystem integration.

**Comment by brandonsavage** [comment]
Gotcha. I'm going to be working on the Flysystem integration soon.
